### PR TITLE
Add string output demo

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: demos/string_output/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# String output model demo {#ovms_string_output_model_demo}
+## Overview
+
+This demo demonstrates usage of OVMS with model that returns string as an output.
+
+### Download Image Net modl
```
Comment (dkalinowski) [inline demos/string_output/README.md:6]:
```suggestion
### Download and prepare Image Net model
```

File: demos/string_output/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# String output model demo {#ovms_string_output_model_demo}
+## Overview
+
+This demo demonstrates usage of OVMS with model that returns string as an output.
```
Comment (dkalinowski) [inline demos/string_output/README.md:4]:
```suggestion
This demo demonstrates example deployment of a model with output precision `ov::element::string`. The output text is serialized into corresponding fields in gRPC proto/REST body. This allows client consume the text directly and skip the process of label mapping or detokenization.
```

File: demos/string_output/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+python3 download_model.py
+```
+
+### Start the OVMS container:
+```bash
+docker run -d -u $(id -u):$(id -g) -v $(pwd):/workspace -p 9178:9178 openvino/model_server:latest \
+--model_path /model --model_name text --rest_port 9178
+```
+
+## Send request
```
Comment (dkalinowski) [inline demos/string_output/README.md:18]: Note which format it is. Also suggest using other available formats gRPC/REST KServe/TFS

File: demos/image_classification_with_string_output/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# String output model demo {#ovms_string_output_model_demo}
+## Overview
+
+This demo demonstrates example deployment of a model with output precision `ov::element::string`. The output text is serialized into corresponding fields in gRPC proto/REST body. This allows the client to consume the text directly and avoid the process of label mapping or detokenization.
+
+### Download and prepare Image Net model using Keras and TensorFlow
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/README.md:6]:
```suggestion
### Download and prepare MobileNet model using from TensorFlow

The script below is downloading a public MobileNet model trained on the ImageNet data. The original model accepts on input the image array in the range of 0-1 and returns probabilities for all the trained classes. We are adding to the model preprocessing function changing the input data range to 0-255 and also postprocessing function which is retrieving the most likely label name as a string. 
This is a very handy functionality because if allows us to export the model with the included pre/post processing functions as the model layers. The client just receives the string data with the label name for the classified image.
```

File: demos/image_classification_with_string_output/README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+```bash
+pip install -r requirements.txt
+python3 download_model.py
+```
+
+### Start the OVMS container:
+```bash
+docker run -d -u $(id -u):$(id -g) -v $(pwd):/workspace -p 9178:9178 openvino/model_server:latest \
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/README.md:15]: use standard ports

File: demos/image_classification_with_string_output/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+--model_path /workspace/model --model_name image_net --rest_port 9178
+```
+
+## Send request
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/README.md:19]: Sending the data with http binary data is not easy with the curl command. It think if would be better just to demonstrate python client.

File: demos/image_classification_with_string_output/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+requests
+numpy
+tensorflow==2.13.1
+keras
+matplotlib
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/requirements.txt:5]: not used

File: demos/image_classification_with_string_output/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+requests
+numpy
+tensorflow==2.13.1
+keras
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/requirements.txt:4]: keras and numpy should be the tf depednency - is it needed?

File: demos/image_classification_with_string_output/README.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
+# String output model demo {#ovms_string_output_model_demo}
+## Overview
+
+This demo demonstrates example deployment of a model with output precision `ov::element::string`. The output text is serialized into corresponding fields in gRPC proto/REST body. This allows the client to consume the text directly and avoid the process of label mapping or detokenization.
+
+### Download and prepare Image Net model using Keras and TensorFlow
+
+```bash
+pip install -r requirements.txt
+python3 download_model.py
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/README.md:10]: It might be handy to show the results of the command: model structure with tree and saved_model_cli or outout of model_metadata calls to ovms to show input and output format.

**Review by dtrawins — CHANGES_REQUESTED** [review]

File: demos/image_classification_with_string_output/image_classification_with_string_output.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        batch_i += 1
+        path = line.strip()
+        with open(path, 'rb') as f:
+            image_data.append(f.read())
+        inputs.append(httpclient.InferInput(args['input_name'], [batch_i], "BYTES"))
+        outputs = []
+        outputs.append(httpclient.InferRequestedOutput(output_name, binary_data=True))
+        
+        nmpy = np.array(image_data , dtype=np.object_)
+        inputs[0].set_data_from_numpy(nmpy)
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/image_classification_with_string_output.py:85]:
```suggestion
        inputs[0].set_data_from_numpy(np.array(image_data , dtype=np.object_))

```

File: demos/image_classification_with_string_output/README.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+```
+Request may be sent also using other APIs (KServ GRPC, TFS). In this sections you can find short code samples how to do this:
+- [TensorFlow Serving API](../../docs/clients_tfs.md)
+- [KServe API](../../docs/clients_kfs.md)
+
+
+## Expected output
+```bash
+Start processing:
+        Model name: image_net
```
Comment (dtrawins) [inline demos/image_classification_with_string_output/README.md:44]: this is not a model name...

File: demos/image_classification_with_string_output/README.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
+# String output model demo {#ovms_string_output_model_demo}
+## Overview
+
+This demo demonstrates example deployment of a model with output precision `ov::element::string`. The output text is serialized into corresponding fields in gRPC proto/REST body. This allows the client to consume the text directly and avoid the process of label mapping or detokenization.
+
+### Download and prepare MobileNet model using from TensorFlow
+The script below is downloading a public MobileNet model trained on the ImageNet data. The original model accepts on input the image array in the range of 0-1 and returns probabilities for all the trained classes. We are adding to the model preprocessing function changing the input data range to 0-255 and also postprocessing function which is retrieving the most likely label name as a string. 
+This is a very handy functionality because if allows us to export the model with the included pre/post processing functions as the model layers. The client just receives the string data with the label name for the classified image.
```
Comment (rasapala) [inline demos/image_classification_with_string_output/README.md:8]:
```suggestion
This is a very handy functionality because it allows us to export the model with the included pre/post processing functions as the model layers. The client just receives the string data with the label name for the classified image.
```

**Review by dkalinowski — APPROVED** [review]

**Review by dtrawins — APPROVED** [review]
